warp/src/main/java/com/schibsted/nmp/warp/components/WarpText.kt

```diff
@@ -0,0 +1,177 @@
+package com.schibsted.nmp.warp.components
+
+import android.content.Context
+import android.util.AttributeSet
+import androidx.annotation.ColorInt
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.AbstractComposeView
+import androidx.compose.ui.semantics.contentDescription
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.unit.dp
+import androidx.core.content.withStyledAttributes
+import com.schibsted.nmp.warp.R
+import com.schibsted.nmp.warp.brands.finn.FinnColors
+import com.schibsted.nmp.warp.components.ext.getTextFromIdOrString
+import com.schibsted.nmp.warp.theme.LocalColors
+import com.schibsted.nmp.warp.theme.LocalTypography
+import com.schibsted.nmp.warp.theme.WarpBrandedTheme
+import com.schibsted.nmp.warp.theme.WarpTheme.colors
+import com.schibsted.nmp.warp.theme.WarpTheme.typography
+import com.schibsted.nmp.warp.utils.FlavorPreviewProvider
+
+@Composable
+fun WarpText(
+    modifier: Modifier = Modifier,
+    text: String,
+    color: Color = colors.primary,
+    style: WarpTextStyle = WarpTextStyle.Body,
+    maxLines: Int = 1,
+) {
+    CompositionLocalProvider(
+        LocalColors provides colors,
+        LocalTypography provides typography,
+    ) {
+
+        val textStyle = getTextStyle(style)
+
+        Text(
+            text = text,
+            modifier = modifier.then(
+                Modifier.semantics(properties = { contentDescription = text })
+            ),
+            overflow = TextOverflow.Ellipsis,
```

Comment:
I think it would be a good idea to add this as an argument with ellipsis as default instead